pkgs/ffigen/lib/src/config_provider/config.dart

```diff
@@ -12,368 +14,731 @@ import 'config_types.dart';
 /// headers.
 // TODO: Add a code snippet example.
 final class FfiGenerator {
-  /// Input config filename, if any.
-  final Uri? filename;
+  /// The configuration for header parsing of [FfiGenerator].
+  final Headers headers;
 
-  /// Path to the clang library.
-  final Uri? libclangDylib;
+  /// Configuration for enums.
+  final Enums enums;
 
-  /// Output file name.
-  final Uri output;
+  /// Configuration for functions.
+  final Functions functions;
 
-  /// Output ObjC file name.
-  final Uri? outputObjC;
+  /// Configuration for globals.
+  final Globals globals;
 
-  Uri get _outputObjC => outputObjC ?? Uri.file('${output.toFilePath()}.m');
+  /// Configuration for macro constants.
+  final Macros macros;
 
-  /// Symbol file config.
-  final SymbolFile? symbolFile;
+  /// Configuration for structs.
+  final Structs structs;
+
+  /// Configuration for typedefs.
+  final Typedefs typedefs;
+
+  /// Configuration for unions.
+  final Unions unions;
+
+  /// Configuration for unnamed enum constants.
+  final UnnamedEnums unnamedEnums;
+
+  /// Objective-C specific configuration.
+  ///
+  /// If `null`, will only generate for C.
+  final ObjectiveC? objectiveC;
+
+  /// Integer types imported from other Dart files.
+  // TODO: Should we move this under `integers: Integers(...)`?
```

Comment:
Do you imagine there'll be more config options on `Integers` in the near future? If not, I would just leave this one top level here.

Reply:
I don't know. But I was thinking just to be consistent with:

```dart
    this.enums = Enums.excludeAll,
    this.functions = Functions.excludeAll,
    this.globals = Globals.excludeAll,
    this.macros = Macros.excludeAll,
    this.structs = Structs.excludeAll,
    this.typedefs = Typedefs.excludeAll,
    this.unions = Unions.excludeAll,
    this.unnamedEnums = UnnamedEnums.excludeAll,
    this.importedIntegers = const <ImportedType>[],
```

Why would we treat integer types different than the others? `Functions` `Structs` and `Macros` have an `imported` field.